src/instructlab/configuration.py

```diff
@@ -586,13 +769,78 @@ def ensure_storage_directories_exist():
         "lora_target_modules": ["q_proj", "k_proj", "v_proj", "o_proj"],
     }
 
+    internal_config = Config(
+        chat=_chat(
+            model=os.path.join(
+                DEFAULTS._cache_home, "models/instructlab/granite-7b-lab"
+            )
+        ),
+        serve=_serve(
+            model_path=os.path.join(
+                DEFAULTS._cache_home, "models/instructlab/granite-7b-lab"
+            ),
+        ),
+        generate=_generate(
+            model=os.path.join(
+                DEFAULTS._cache_home, "models/mistralai/mixtral-8x7b-v0-1"
+            ),
+            gpus=4,
+            teacher=_serve(
+                backend="vllm",
+                model_path=os.path.join(
+                    DEFAULTS._cache_home, "models/mistralai/mixtral-8x7b-v0-1"
+                ),
+            ),
+        ),
+        train=_train(
+            additional_args={
+                "warmup_steps": 10,
+                "learning_rate": 2e-6,
+                "lora_dropout": 0.1,
+                "lora_alpha": 32,
+            },
+            ckpt_output_dir=os.path.join(DEFAULTS._data_dir, "checkpoints"),
+            data_output_dir=os.path.join(DEFAULTS._data_dir, "internal"),
+            data_path=os.path.join(DEFAULTS._data_dir, "datasets"),
+            nproc_per_node=4,
+            effective_batch_size=96,
+            lora_quantize_dtype="nf4",
+            max_batch_len=20000,
+            max_seq_len=4096,
+            save_samples=1000,
+            deepspeed_cpu_offload_optimizer=True,
+            deepspeed_cpu_offload_optimizer_ratio=1,
+            deepspeed_cpu_offload_optimizer_pin_memory=False,
+            is_padding_free=True,
+            model_path=os.path.join(
+                DEFAULTS._cache_home, "models/instructlab/granite-7b-starter"
+            ),
+        ),
+        evaluate=_evaluate(
+            base_model=os.path.join(
+                DEFAULTS._cache_home, "models/instructlab/granite-7b-starter"
+            ),
+            gpus=4,
+            mt_bench=_mtbench(
+                judge_model=os.path.join(
+                    DEFAULTS._cache_home, "models/prometheus-eval/prometheus-8x7b-v2.0"
+                ),
+                max_workers=96,
+            ),
+        ),
+    )
+
     # create exper_args file for users to see/edit
     if not os.path.isfile(DEFAULTS.TRAIN_ADDITIONAL_OPTIONS_FILE):
         with open(
             DEFAULTS.TRAIN_ADDITIONAL_OPTIONS_FILE, "w", encoding="utf-8"
         ) as outfile:
             yaml.dump(additional_args_and_defaults, outfile)
 
+    if not os.path.isfile(DEFAULTS.INTERNAL_CONFIG_FILE):
```

Comment:
I thought the plan was to not generate these in-code?..

Reply:
no, this is still the plan as of the last conversation I believe